virtual_rainforest/models/animals/animal_cohorts.py

```diff
@@ -273,24 +275,37 @@ def forage_cohort(
 
         """
 
-        if self.functional_group.diet == DietType.HERBIVORE and plant_list:
-            food_choice = choice(plant_list)
-            consumed_energy = self.eat(food_choice, excrement_pool)
-        elif self.functional_group.diet == DietType.CARNIVORE and animal_list:
-            food_choice = choice(animal_list)
-            consumed_energy = self.eat(food_choice, carcass_pool)
-        else:
-            LOGGER.info("No food available.")
+        mass_consumed = 0.0  # Initialize to 0.0
+
+        try:
+            if self.functional_group.diet == DietType.HERBIVORE and plant_list:
+                food_choice = choice(plant_list)
+                mass_consumed = self.eat(food_choice, excrement_pool)
+            elif self.functional_group.diet == DietType.CARNIVORE and animal_list:
+                food_choice = choice(animal_list)
+                mass_consumed = self.eat(food_choice, carcass_pool)
+            else:
+                LOGGER.info("No food available.")
+                food_choice = None  # No food available
+
+        except ValueError as e:
+            if str(e) == "Individuals cannot be 0.":
+                LOGGER.warning("Tried to eat with zero individuals in the cohort.")
+                mass_consumed = 0
+                food_choice = None  # No food was actually consumed
+            else:
+                raise
```

Comment:
I'm not convinced by this approach. Using the _text_ of a exception to decide what to do seems a bit hacky to me.

My personal preference would be to check if individuals are zero in the appropriate function and then return the correct value for `food_choice` and `mass_consumed`. If that is deemed to complicated to implement, then create a custom exception, eg. `EatingZeroIndividualsCohortError`, catch that (and only that) to act accordingly, leaving any other exception unhandled.

Reply:
Ah, I see what you mean! I'll get that worked out.